### PR TITLE
blog: experiment refs post

### PR DESCRIPTION
**Pull request by pmrowla**
Currently the publish date is set to next Monday (4/19) - I'm assuming we don't want it to go up on a Friday.

~~This also still needs a comments thread opened, but everything else should be ready to merge next week~~

Should be ready to merge

### CODE REVIEW

File: content/blog/2021-04-19-experiment-refs.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
+---
+title: Utilizing Custom Git References in DVC
+date: 2021-04-19
+description: |
+  In DVC 2.0, we’ve introduced a new feature set aimed at simplifying the
+  versioning of lightweight ML experiments. In this post, we’ll dive into
+  how exactly these new experiments work.
+
+descriptionLong: |
+  In DVC 2.0, we’ve introduced a new feature set aimed at simplifying the
```
Comment (shcheklein) [inline content/blog/2021-04-19-experiment-refs.md:10]: 💅  it supports links and regular Markdown text styles. Should we link to the previous blog post, the release 2.0?

File: content/blog/2021-04-19-experiment-refs.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+Even though the experiment refs themselves are “invisible” to Git porcelain
+commands, Git commit SHAs for experiments can be used in any Git command. This
+allows you to leverage the power of tools like `git diff` to compare things like
+code changes between a DVC experiment and any other Git commit (meaning you can
+even compare experiment commit SHAs to Git branches or tags).
+
+Likewise, for tools which provide a GUI on top of Git, experiments will be
+hidden from your repository in typical use cases:
+
+![`gitk --branches --tags` example](/uploads/images/2021-04-19/gitk-branches-tags.png 'gitk --branches --tags')
```
Comment (shcheklein) [inline content/blog/2021-04-19-experiment-refs.md:246]: - images are too small (on retina screens?) - can we make a high res large screenshots?
- to make `gitk --branches --tags` a subtitle - put after the `) _gitk --branches --tags_` ?

**Review by shcheklein — CHANGES_REQUESTED** [review]
static/uploads/images/2021-04-19/experiment-refs.png - is huge - let's run minify it

**Comment by shcheklein** [comment]
@pmrowla really great work! thank you.

File: content/blog/2021-04-19-experiment-refs.md (hunk 2 above)
Comment (dmpetrov) [reply content/blog/2021-04-19-experiment-refs.md:246]: 💯 
I see nothing in the original one 🤓  and it is hard to read even if I click to it.

File: content/blog/2021-04-19-experiment-refs.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
+---
+title: Utilizing Custom Git References in DVC
```
Comment (dmpetrov) [inline content/blog/2021-04-19-experiment-refs.md:2]: Should we include ML experiment in the title to make it more general?
Like `Git Custom References for ML experiments`

File: content/blog/2021-04-19-experiment-refs.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+specified parameter value. DVC then reproduced our pipeline as if we had
+manually edited our `params.yaml` to contain that parameter change (setting
+`featurize.max_features` to `2000`), and then saved the results in a new
+experiment named `exp-9b1a5`.
+
+_Note: `--temp` is an optional flag for `dvc exp run` which makes DVC run the
+experiment in a temporary directory instead of in our main workspace. We use
+`--temp` here to illustrate how DVC tracks experiment changes without needing to
+modify the workspace at all, but experiments can also be run directly in your
+workspace as well (like in traditional `dvc repro` workflows)._
```
Comment (dmpetrov) [inline content/blog/2021-04-19-experiment-refs.md:77]: Minor: I don't see the value of running it in temp dir. We can keep it is as.

File: content/blog/2021-04-19-experiment-refs.md [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+namespaces) can also be used to view experiments as if they were Git branches:
+
+![`gitk --all` example screenshot](/uploads/images/2021-04-19/gitk-all.png 'gitk --all')
+
+Experiments are also completely local (since custom refs are not transferred to
+or from Git remotes on `git push` and `git pull`), meaning that even if you run
+thousands of experiments locally, you do not need to worry about accidentally
+polluting your team’s upstream Github or Gitlab repository with those
+experiments. However, individual DVC experiments can be explicitly shared via
+remote Git repositories using the `dvc exp push` and `dvc exp pull` commands.
```
Comment (dmpetrov) [inline content/blog/2021-04-19-experiment-refs.md:258]: `dvc exp branch` should be mentioned here as a default way of materializing the ephemeral experiment.

**Review by dmpetrov — APPROVED** [review]
Great post @pmrowla!
A couple of minor suggestions are inline.

File: content/blog/2021-04-19-experiment-refs.md (hunk 3 above)
Comment (pmrowla) [reply content/blog/2021-04-19-experiment-refs.md:2]: I'm fine with changing the title but we'd need a new cover image @jendefig

File: content/blog/2021-04-19-experiment-refs.md (hunk 4 above)
Comment (pmrowla) [reply content/blog/2021-04-19-experiment-refs.md:77]: The main reasoning was that some extraneous internal DVC-only refs will show up in the low-level git commands if we demo with workspace runs. These refs are kept in the execution environment only, so for `--temp` runs they would not show up in our demo commands.

But I don't think it's a big deal either way, so I can adjust the demo to do workspace runs

**Comment by pmrowla** [comment]
Changes for all the review comments have been made other than the title (cover image will need to be updated)

File: content/blog/2021-04-19-experiment-refs.md (hunk 3 above)
Comment (dmpetrov) [reply content/blog/2021-04-19-experiment-refs.md:2]: This one looks good to me. It should be 100% match. I'm fine with the original one if you think DVC is important.

An idea if you prefer keep DVC - `DVC: Git Custom References for ML experiments`

**Comment by shcheklein** [comment]
@pmrowla do we want to publish it now?

**Comment by pmrowla** [comment]
> @pmrowla do we want to publish it now?

@shcheklein I guess it's getting late enough in the US that it's probably better to wait until business hours tomorrow?

Either way, just wanted to make it clear that we don't need to wait for any other changes now, and it can be published at any time.

**Comment by shcheklein** [comment]
@pmrowla sounds good! @jendefig please let me know when we are ready to publish and promote it. Will we need to adjust the dates though?

**Comment by pmrowla** [comment]
I don't think it's necessary to change the actual paths for the static images or the post filename itself?

The `date` metadata field at the top of the post markdown can be adjusted as needed, anyone with write perms on this repo should be able to change it in the PR right before it gets published.

**Comment by shcheklein** [comment]
> I don't think it's necessary to change the actual paths for the static images or the post filename itself?

I don't remember to be honest. I would try to keep it consistent to be honest. It should be pretty quick anyway. I can do this before publishing, let's agree on time when we do this.

**Comment by jendefig** [comment]
@shcheklein @pmrowla I'm good with this going asap.  I planned to promote tomorrow.  Waiting for the link so I can load in Buffer. And will be in the Wednesday email.

**Comment by shcheklein** [comment]
@jendefig @pmrowla done, merged!

File: content/blog/2021-04-19-experiment-refs.md [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+[2.0 release post](/blog/dvc-2-0-release) and our
+[Get Started: Experiments](/doc/start/experiments) guide._
+
+## Experiments in DVC 2.0
+
+At the heart of the new experiments feature is the `dvc exp run` command.
+Whenever a pipeline is executed with `dvc exp run`, the results will be
+automatically tracked by DVC as a single “experiment”. DVC will track everything
+in your workspace as a part of the experiment, including unstaged changes made
+prior to execution. This means that DVC experiments can be used to test the
```
Comment (jorgeorpinel) [inline content/blog/2021-04-19-experiment-refs.md:55]: > including unstaged changes

Except totally new files, unless Git-staged (see https://github.com/iterative/dvc/pull/5029). Prob. too deep to include, anyway.

File: content/blog/2021-04-19-experiment-refs.md [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+result of changes to DVC-tracked data or pipeline parameters, as well as changes
+to Git-tracked code.
+
+![Example experiment run](/uploads/images/2021-04-19/exp-run.gif 'Example experiment run')
```
Comment (jorgeorpinel) [inline content/blog/2021-04-19-experiment-refs.md:59]: The exp run takes a bit but the results (2nd `exp show` table) is only shown for like a second before the GIF restarts.

**Review by jorgeorpinel — COMMENTED** [review]
Just 2 comments post-merge. Probably no action needed. Great read BTW!

**Pull request merged**